api/envoy/config/listener/v3/quic_config.proto

```diff
@@ -86,4 +92,7 @@ message QuicProtocolOptions {
   // If not specified, no debug visitor will be attached to connections.
   // [#extension-category: envoy.quic.connection_debug_visitor]
   core.v3.TypedExtensionConfig connection_debug_visitor_config = 11;
+
+  // Configure a type of UDP cmsg to pass to listener filters via QuicReceivedPacket.
```

Comment:
Correct.

Reply:
Since Envoy APIs tend to hang around stone forever, would we every want to save more than one CMSG type? I know we have code that reads the TOS byte to get ECN bits, for example. So maybe we might want to use this mechanism to save this data? Also, for ECN the level is IPPROTO_IP for IPv4 and IPPROTO_IPV6 for IPv6, which might suggest a need for both levels to be specified?

Reply:
Right now we're taking advantage of the so-far unused `packet_headers` and `headers_length` fields in [QuicReceivedPacket](https://github.com/envoyproxy/envoy/blob/dde04ae6c6c3b5af191fbe16aa3b27d91226bafc/source/common/quic/active_quic_listener.cc#L140C54-L140C61) which currently allows for one header, so it'd require a bigger refactor. I agree it seems like a possible need in the future. I'd say we can come back to that if/when we need? Let me know what you think.